content/docs/run-menu.md

```diff
@@ -0,0 +1,85 @@
+
+### Run your makefile
+
+If you are editing source code, you might want to have a shortcut that will run `make` or `nmake` or `gmake` or similar build program in the current directory.
+
+### Getting External Help
+
+You could define **Run** menu entries to allow you to get help.  For example, running the command `https://en.wikipedia.org/wiki/Special:Search?search=$(CURRENT_WORD)` will look up the currently selected word (or the word where your caret is if you don't have a selection) in Wikipedia using your default browser.  Or running `https://docs.python.org/3/search.html?check_keywords=yes&amp;q=$(CURRENT_WORD)` will look up the current word in the Python 3 online documentation.
+
+### A Filename in Your Text
+
+If your cursor or selection is on the name of a file in your text, you can open that file in the current Notepad++ instance by running:
+```
+"$(NPP_DIRECTORY)\notepad++.exe" "$(CURRENT_WORD)"
+```
+
+Or, if you want to force it to open in a new instance, use:
+```
+"$(NPP_DIRECTORY)\notepad++.exe" "$(CURRENT_WORD)" -nosession -multiInst
+```
+
+### You Are Limited Only By Your Imagination
+
+If you can run it from `cmd.exe` or `powershell.exe` in a single command line, you can run it from here.  Let your imagination run wild.
+
+## Going Beyond
+
+If a single-line command isn't sufficient for your needs, you may want to consider one of the following options:
+
+1. Write a batch file: writing a windows `.bat` or `.cmd` or `.ps1` file using cmd.exe or powershell syntax will allow you to specify a group of commands; you can then just use the **Run > Run...**  to call that batch file to do your more complicated task.  You can pass any of the [special variables](https://npp-user-manual.org/docs/config-files/#userdefinedcommands) as arguments to this batch file.
+
+2. Use the [NppExec plugin](https://github.com/d0vgan/nppexec/): this plugin can be installed from the **Plugins > Plugins Admin** interface, and gives you access to a custom batch language, but with extended features that give you access to all the [special variables](https://npp-user-manual.org/docs/config-files/#userdefinedcommands) plus extra access to the internals of the Notepad++ interface.  This also allows you to view the output of commands in an embedded interactive console window that can be docked in Notepad++.
+
+3. Use [PythonScript](https://github.com/bruderstein/PythonScript) or [LuaScript](https://github.com/dail8859/LuaScript) or [jN Notepad++](https://github.com/sieukrem/jn-npp-plugin/wiki) plugins (or similar) to write a script in your favorite programming language (and all those languages should give you access to any feature of that language, plus a way to access applications that live on your filesystem).  These may provide interactive console windows to give you even more flexibility.
```

Comment:
Should it be made clearer that point 2 and 3 here do not involve the Run menu at all?  If point 1 didn't mention it, the title of the section could be "Going Beyond the Run menu"....  (I was trying to think like a newbie).